examples/demo/components/director/director.go

```diff
@@ -0,0 +1,150 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package director
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"open-match.dev/open-match/examples/demo/components"
+	"open-match.dev/open-match/internal/rpc"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+func Run(ds *components.DemoShared) {
+	for !isContextDone(ds.Ctx) {
+		run(ds)
+	}
+}
+
+func isContextDone(ctx context.Context) bool {
+	select {
+	case <-ctx.Done():
+		return true
+	default:
+		return false
+	}
+}
+
+type status struct {
+	Status        string
+	LatestMatches []*pb.Match
+}
+
+func run(ds *components.DemoShared) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			err, ok := r.(error)
+			if !ok {
+				err = fmt.Errorf("pkg: %v", r)
+			}
+
+			ds.Update(status{Status: fmt.Sprintf("Encountered error: %s", err.Error())})
+			time.Sleep(time.Second * 10)
+		}
+	}()
+
+	s := status{}
+
+	//////////////////////////////////////////////////////////////////////////////
+	s.Status = "Connecting to backend"
+	ds.Update(s)
+
+	conn, err := rpc.GRPCClientFromConfig(ds.Cfg, "api.backend")
+	if err != nil {
+		panic(err)
+	}
+	defer conn.Close()
+	be := pb.NewBackendClient(conn)
+
+	//////////////////////////////////////////////////////////////////////////////
+	s.Status = "Match Match: Sending Request"
+	ds.Update(s)
+
+	var matches []*pb.Match
+	{
+		req := &pb.FetchMatchesRequest{
+			Config: &pb.FunctionConfig{
+				Host: "om-demofunction",
+				Port: 50502,
+				Type: pb.FunctionConfig_GRPC,
+			},
+			Profile: []*pb.MatchProfile{
+				&pb.MatchProfile{
+					Name: "1v1",
+					Pool: []*pb.Pool{
+						{
+							Name: "Everyone",
+							Filter: []*pb.Filter{
+								&pb.Filter{
+									Attribute: "mode.demo",
+									Min:       -100,
+									Max:       100,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		resp, err := be.FetchMatches(ds.Ctx, req)
+		if err != nil {
+			panic(err)
+		}
+
+		matches = resp.Match
+	}
+
+	//////////////////////////////////////////////////////////////////////////////
+	s.Status = "Matches Found"
+	s.LatestMatches = matches
+	ds.Update(s)
+
+	//////////////////////////////////////////////////////////////////////////////
+	s.Status = "Assigning Players"
+	ds.Update(s)
+
+	for _, match := range matches {
+		ids := []string{}
+
+		for _, t := range match.Ticket {
+			ids = append(ids, t.Id)
+		}
+
+		req := &pb.AssignTicketsRequest{
+			TicketId: ids,
+			Assignment: &pb.Assignment{
+				Connection: fmt.Sprintf("%d.%d.%d.%d:2222", rand.Intn(256), rand.Intn(256), rand.Intn(256), rand.Intn(256)),
+			},
+		}
+
+		resp, err := be.AssignTickets(ds.Ctx, req)
+		if err != nil {
+			panic(err)
+		}
+
+		_ = resp
```

Comment:
```suggestion
```

Reply:
If I don't include this line, resp becomes unused, and throws an error.  I don't want not create resp in the first place because I want to be explicit here that we're doing nothing with the response.